packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -39,90 +86,199 @@ const ConfirmSignupCode = ({
     submitButtonText: 'Confirm',
   };
 
-  const handleResendCode = () => {
-    // TODO: add resend code action
-    // account.verifySessionResendCode()
-    // if success, display alert bar message
-    // 'Email resent. Add accounts@firefox.com to your contacts to ensure a smooth delivery.'
-  };
+  const localizedCustomCodeRequiredMessage = ftlMsgResolver.getMsg(
+    'confirm-signup-code-is-required-error',
+    'Confirmation code is required'
+  );
+
+  // When the user types in the code input field, all banners and tooltips should be cleared
+  // Timeout is added to reduce jankiness, but does not include a smooth hiding effect.
+  useEffect(() => {
+    if (clearMessages) {
+      const timer = setTimeout(() => {
+        setCodeErrorMessage('');
+        setBanner({ visible: false, type: undefined, message: undefined });
+        setClearMessages(false);
+      }, 750);
+      return () => clearTimeout(timer);
+    }
+    return;
```

Comment:
Do we need this `return`?

Reply:
Maybe there's a way around it, but I was getting an error that not all paths returned a value.